scripts/rust_file.py

```diff
@@ -77,12 +77,15 @@ def __str__(self) -> str:
     def __hash__(self) -> int:
         return hash((self.visibility, self.name))
 
-    def __eq__(self, other: "RustMod") -> bool:
-        return self.name == other.name and self.visibility == other.visibility
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, RustMod):
+            return self.name == other.name and self.visibility == other.visibility
+        else:
+            return False
```

Comment:
```suggestion
        return isinstance(other, RustMod) and self.name == other.name and self.visibility == other.visibility
```

Does this still type-check?  If so, I think it's preferred.